Alliance.Server/Extensions/WargAttack/Behavior/WargBehavior.cs

```diff
@@ -405,96 +409,32 @@ private void ChaseTarget()
-				Vec3 offset = new(Agent.GetMovementDirection() * 10);
-				destination = (Agent.Position + offset).ToWorldPosition();
-			}
-			// If far from target, randomly make a detour
-			else if (distanceToTarget > 8)
-			{
-				Vec3 offset = new(_target.GetMovementDirection() * distanceToTarget);
-				offset.RotateAboutZ(MBRandom.RandomInt(-2, 2));
-				destination = (_target.Position + offset).ToWorldPosition();
-			}
-			//Log($"Warg chasing target to {targetPos.GetGroundVec3()} ({_chaseTrajectory})", LogLevel.Debug);
-			Agent.SetScriptedPosition(ref destination, false, Agent.AIScriptedFrameFlags.None);
-			return;
-
-
-
-
-			// TODO remove below if above code is satisfying
-
-			// Set speed depending on distance to target
-			if (distanceToTarget > WargConstants.CHASE_RADIUS / 2 || MBRandom.RandomFloat < 0.25f) // 25% chance to charge, or if far from target
-			{
-				Agent.SetMaximumSpeedLimit(Agent.Monster.WalkingSpeedLimit * 3f, false); // Charge speed
-			}
-			else
-			{
-				Agent.SetMaximumSpeedLimit(Agent.Monster.WalkingSpeedLimit, false); // Slow closing in
-			}
+			Agent.SetMaximumSpeedLimit(Agent.Monster.WalkingSpeedLimit * 5f, false); // Charge speed
 
-			// Adjust facing direction to keep track of the target
-			Vec3 directionToTarget = (_target.Position - Agent.Position).NormalizedCopy();
-			Vec3 curveOffset = Vec3.Zero;
-			Vec3 extensionBeyondTarget = Vec3.Zero; // Extend the position beyond the target
+			Vec3 estimatedTargetDestination = _target.Position + (_target.GetMovementDirection() * _target.MovementVelocity.Y * 1).ToVec3();
+			Vec3 directionToEstimatedTarget = (estimatedTargetDestination - Agent.Position).NormalizedCopy();
+			Vec3 positionToChase = estimatedTargetDestination + directionToEstimatedTarget * 5f;
 
-			// Define curve offset and extension beyond target based on chosen trajectory
-			switch (_chaseTrajectory)
-			{
-				case ChaseTrajectory.CurveLeft:
-					curveOffset = new Vec3(-directionToTarget.y, directionToTarget.x, 0) * distanceToTarget / 5; // Modified to curve left correctly
-					break;
-				case ChaseTrajectory.CurveRight:
-					curveOffset = new Vec3(directionToTarget.y, -directionToTarget.x, 0) * distanceToTarget / 5; // Modified to curve right correctly
-					break;
-				case ChaseTrajectory.Straight:
-					break;
-			}
+			WorldPosition destination = positionToChase.ToWorldPosition();
 
-			if (WargAttackHelper.IsInFrontCone(Agent, _target, 30))
-			{
-				extensionBeyondTarget = directionToTarget * (1 + distanceToTarget + Agent.MovementVelocity.Y);
-			}
-			else
+			// If target is behind the warg, get some distance before turning around
+			bool isTargetInFront = WargAttackHelper.IsInFrontCone(Agent, _target, 180);
+			if (distanceToTarget < 8 && !isTargetInFront)
 			{
-				extensionBeyondTarget = -directionToTarget;
+				// Just go straight to gain distance
+				destination = (Agent.Position + (Agent.LookDirection * 100f)).ToWorldPosition();
```

Comment:
The destination will be reached using navmesh, so the warg will still turn and avoid obstacles. I'm not sure if it can cause the warg to be stuck 🤔 But we should keep an eye on it for sure

Reply:
Yep that is a possibility if no path exist to the destination. We would need a way to identify if destination can be physically reached to prevent these.